packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+   * Label to be rendered with the checkbox.
+   */
+  label?: ShorthandProps<LabelProps>;
+
+  /**
+   * Icon to be displayed when the checkbox is in the checked state.
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
+
+  /**
+   * A checkbox can be rendered with a circular shape.
+   */
+  circular?: boolean;
+
+  /**
+   * A checkbox's state can be controlled.
+   * @defaultvalue false
+   */
+  checked?: 'indeterminate' | boolean;
+
+  /**
+   * Whether the checkbox should be rendered as checked by default.
+   */
+  defaultChecked?: 'indeterminate' | boolean;
+
+  /**
+   * Checkbox supports two different checkbox sizes.
+   * @defaultvalue 'medium'
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Determines whether the label should be positioned before (start) or after (end) the checkbox.
+   * @defaultvalue 'end'
+   */
+  labelPosition?: 'start' | 'end';
+
+  /**
+   * ID of the root element that wraps the checkbox and label.
+   */
+  rootId?: string;
+
+  /**
+   * ID of the <input/> element that represents the checkbox.
+   */
+  id?: string;
+
+  /**
+   * Callback to be called when the checked state value changes.
+   */
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: CheckboxOnChangeData) => void;
+}
+
+/**
+ * Data for the onChange event for checkbox.
+ */
+export interface CheckboxOnChangeData {
+  checked?: 'indeterminate' | boolean;
+}
+```
+
+Some feedback needed for props are:
+
+- `checked` will work similar to `aria-checked` and have `true`, `false`, and `indeterminate` options.
+- A `rootId` is needed since the input and label will share an id and `rootId` will be used for the root element.
+- To initially make the checkbox simple, the icon will be a `ReactElement` instead of a slot.
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### **Public**
 
-## Migration
+```tsx
+<Checkbox label="Example Checkbox" />
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+### **Internal**
 
-- _Migration from v8_
-- _Migration from v0_
+```tsx
+<slots.root {...slotProps.root}>
+  {state.labelPosition === 'start' && <slots.label {...slotProps.label} />}
+  <div className={state.checkboxClassName}>
+    <div className={state.iconClassName}>{icon}</div>
+    <input
+      type="checkbox"
+      className={state.inputClassName}
+      ref={state.inputRef}
+      id={state.inputId}
+      onChange={state.inputOnChange}
+    />
+  </div>
+  {state.labelPosition === 'end' && <slots.label {...slotProps.label} />}
+</slots.root>
+```
 
-## Behaviors
+### **DOM**
+
+In this case, the label position is set to end as per default.
+
+```tsx
+<div {/* Root Element */}>
+  <div {/* Checkbox */}>
+    <div {/* Icon */}>
+      ...Icon
+    </div>
+    <input id="cbox-1" type="checkbox" aria-checked="..." aria-label="..." aria-labelledby="..."
```

Comment:
The native `input` checkbox supports `checked` and `indeterminate` IDL attributes, can we avoid using `aria-checked` in this case ? 
```suggestion
    <input id="cbox-1" type="checkbox" aria-label="..." aria-labelledby="..."
```

Reply:
After a sync with cxe-red we decided to now use `mixed` instead of `indeterminate` to follow aria-checked and set the indeterminate attribute as well.